quixstreams/sinks/community/mqtt.py

```diff
@@ -0,0 +1,169 @@
+        super().__init__()
+
+        self.mqtt_version = mqtt_version
+        self.mqtt_username = mqtt_username
+        self.mqtt_password = mqtt_password
+        self.mqtt_topic_root = mqtt_topic_root
+        self.tls_enabled = tls_enabled
+        self.qos = qos
+
+        self.mqtt_client = paho.Client(
+            callback_api_version=paho.CallbackAPIVersion.VERSION2,
+            client_id=mqtt_client_id,
+            userdata=None,
+            protocol=self._mqtt_protocol_version(),
+        )
+
+        if self.tls_enabled:
+            self.mqtt_client.tls_set(
+                tls_version=mqtt.client.ssl.PROTOCOL_TLS
+            )  # we'll be using tls now
+
+        self.mqtt_client.reconnect_delay_set(5, 60)
+        self._configure_authentication()
+        self.mqtt_client.on_connect = self._mqtt_on_connect_cb
+        self.mqtt_client.on_disconnect = self._mqtt_on_disconnect_cb
+        self.mqtt_client.connect(mqtt_server, int(mqtt_port))
+
+    # setting callbacks for different events to see if it works, print the message etc.
+    def _mqtt_on_connect_cb(
+        self,
+        client: paho.Client,
+        userdata: any,
+        connect_flags: paho.ConnectFlags,
+        reason_code: paho.ReasonCode,
+        properties: paho.Properties,
+    ):
+        if reason_code == 0:
+            print("CONNECTED!")  # required for Quix to know this has connected
+        else:
+            print(f"ERROR ({reason_code.value}). {reason_code.getName()}")
+
+    def _mqtt_on_disconnect_cb(
+        self,
+        client: paho.Client,
+        userdata: any,
+        disconnect_flags: paho.DisconnectFlags,
+        reason_code: paho.ReasonCode,
+        properties: paho.Properties,
+    ):
+        print(
+            f"DISCONNECTED! Reason code ({reason_code.value}) {reason_code.getName()}!"
+        )
+
+    def _mqtt_protocol_version(self):
+        if self.mqtt_version == "3.1":
+            return paho.MQTTv31
+        elif self.mqtt_version == "3.1.1":
+            return paho.MQTTv311
+        elif self.mqtt_version == "5":
+            return paho.MQTTv5
+        else:
+            raise ValueError(f"Unsupported MQTT version: {self.mqtt_version}")
+
+    def _configure_authentication(self):
+        if self.mqtt_username:
+            self.mqtt_client.username_pw_set(self.mqtt_username, self.mqtt_password)
+
+    def _publish_to_mqtt(
+        self,
+        data: str,
+        key: bytes,
+        timestamp: datetime,
+        headers: List[Tuple[str, HeaderValue]],
+    ):
+        if isinstance(data, bytes):
+            data = data.decode("utf-8")  # Decode bytes to string using utf-8
+
+        json_data = json.dumps(data)
+        message_key_string = key.decode(
+            "utf-8"
+        )  # Convert to string using utf-8 encoding
+        # publish to MQTT
+        self.mqtt_client.publish(
+            self.mqtt_topic_root + "/" + message_key_string,
+            payload=json_data,
+            qos=self.qos,
```

Comment:
According to [docs](https://eclipse.dev/paho/files/paho.mqtt.python/html/client.html#paho.mqtt.client.Client.publish), it would be great to have a control over `retain` and `properties` arguments.
Ideally the constructor should accept either constant or callbacks
```python
retain: bool | Callable[[Any], bool] = False,
properties: paho.mqtt.properties.Properties | Callable[[Any], paho.mqtt.properties.Properties] | None = None,
```